packages/babel-helper-define-polyfill-provider/src/types.ts

```diff
@@ -0,0 +1,120 @@
+import type { NodePath } from "@babel/traverse";
+import { types as t } from "@babel/core";
+import createMetaResolver from "./meta-resolver";
+
+type ObjectMap<T> = {
+  [k: string]: T;
+};
+
+export type Pattern = string | RegExp;
+
+export type MissingDependenciesOption =
+  | false
+  | {
+      log?: "per-file" | "deferred";
+      inject?: "import" | "throw";
+      // When true, log all the polyfills without checking if they are installed
+      all?: boolean;
+    };
+
+export type ProviderOptions<Opts = {}> = {
+  include?: Pattern[];
+  exclude?: Pattern[];
+} & Opts;
+
+export type PluginOptions = {
+  method: MethodString;
+  targets?: {
+    browsers: string | string[];
+    [target: string]: string | number;
```

Comment:
This is correct, because you can have things like `node: 16` or `node: "16.7"`.

Reply:
I meant that `browsers` and `target` seem to be incompatible with each other (removing one or the other resolves the type error).

The type error (can see in CI logs) is basically complaining that `string[]` is not an `index` type

Reply:
For this to type-check, it would have to be:

```suggestion
    browsers: string | string[];
    [target: string]: string | string[] | number;
```

Since TS requires the index signature [to be a union](https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures) of all the existing types

That actually changes the code, so I'm not really sure what to change this to as I'm not sure what was meant to be expressed

Reply:
See follow-up discussion in https://github.com/babel/babel-polyfills/pull/121#discussion_r838033963